exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillLimitRelBase.java

```diff
@@ -36,11 +36,17 @@
 public abstract class DrillLimitRelBase extends SingleRel implements DrillRelNode {
   protected RexNode offset;
   protected RexNode fetch;
+  private boolean pushDown;  // whether limit has been push past its child.
```

Comment:
LIMIT is somewhat unique operator in that even when it is pushed past the child, the original LIMIT still remains.  Can you add that in the comments because the pushDown flag will be TRUE for the original LIMIT and FALSE for the pushed version.